include/bout/petsclib.hxx

```diff
@@ -109,11 +119,13 @@ private:
 
 class PetscLib {
 public:
-  PetscLib() {}
+  PetscLib(Options* UNUSED(opt) = nullptr) {}
   ~PetscLib() {}
   
   static void setArgs(int &UNUSED(c), char** &UNUSED(v)) {}
   
+  KSP createKSPWithOptions(MPI_Comm& comm);
```

Comment:
Is `KSP` going to be defined if we're building without petsc or do we need to do something like `define KSP = void*`?

Also without going into detail it looks like we could avoid needing two different definitions of `PetscLib` in this header if we just chopped out the implementations in the source file (which I guess is what happens for the createKsp routine, I've not gotten that far yet!)

Reply:
I'm not sure. As it is at the moment, I should have put an empty body into the no-PETSc branch of the header; I just missed it...

Reply:
Good point, yes, we'll need a definition of `KSP` (although with `using KSP = void*`, but yes).

I think `class KSP;` might even be sufficient because nothing should be calling `createKSPWithOptions` without PETSc linked (that is, we should be `#ifdef`-ing the calls out)